cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/BrokerFailures.java

```diff
@@ -20,13 +19,14 @@
 import static com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig.BROKER_FAILURE_EXCLUDE_RECENTLY_REMOVED_BROKERS_CONFIG;
 import static com.linkedin.kafka.cruisecontrol.detector.AnomalyDetectorUtils.getSelfHealingGoalNames;
 import static com.linkedin.kafka.cruisecontrol.detector.AnomalyDetectorUtils.KAFKA_CRUISE_CONTROL_OBJECT_CONFIG;
+import static com.linkedin.kafka.cruisecontrol.detector.notifier.AnomalyType.BROKER_FAILURE;
 
 
 /**
  * The broker failures that have been detected.
  */
 public class BrokerFailures extends KafkaAnomaly implements CruiseControlConfigurable {
-  protected static final String ID_PREFIX = AnomalyType.BROKER_FAILURE.toString();
```

Comment:
Now that we have a separation between the following:
```
  TRIGGERED_USER_TASK_ID = "triggeredUserTaskId";
  TRIGGERED_SELF_HEALING_TASK_ID = "triggeredSelfHealingTaskId";
  TRIGGERED_TASK_REASON = "triggeredTaskReason";
```

I am curious if we can drop `ID_PREFIX` overwrite from individual anomaly types (and handle the drop accordingly in the rest of the codebase) -> thus have an `anomalyId` that is a proper `uuid`?

Reply:
we have some places in our code depend on the assumption that self-healing task's uuid starts with anomaly type. 

One example is [determine whether mark `AnomalyDetector` or `UserTaskManager` when execution starts](https://github.com/linkedin/cruise-control/blob/d10aa73209ce0f542afca44ad7145e7b8fe8112d/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java#L699).
If we remove the prefix, we need to make decision based on `_reason` field, we can chose a reason template for self-healing tasks; but for user-triggered task, the `_reason` is provided by user, so theoretically user can accidentally provide a reason which matchs our pre-defined self-healing reason template.

Reply:
Why not change the places that depends on `uuid` prefix? For the same reason, I don't think we should rely on parsing the `_reason` field.

For the specific example above, can't we just pass the information regarding `who` started the execution? Internally, we always know whether the execution was triggered by self healing or not; hence, it is possible to pass this information down to the executor.

Reply:
That's a good idea! 
I removed the prefix  in UUID and pass down the information of who triggers operation from `Runnable` to `Executor`.